src/rabbit_amqqueue.erl

```diff
@@ -274,9 +274,15 @@ declare(QueueName, Durable, AutoDelete, Args, Owner, Node) ->
                                       recoverable_slaves = [],
                                       gm_pids            = [],
                                       state              = live})),
-    Node = rabbit_mirror_queue_misc:initial_queue_node(Q, Node),
+
+    Node1 = case rabbit_queue_master_location_misc:get_location(Q)  of
+              {ok, Node0}  -> Node0;
+              {error, _}   -> Node
+            end,
+
+    Node1 = rabbit_mirror_queue_misc:initial_queue_node(Q, Node1),
```

Comment:
OK, so `ha-params` effects on this change have been discussed out-of-band. The conclusion is: `ha-params` with specific nodes listed currently overrides the locator. This is fine: using them at the same time makes little sense. We'll mention that in the docs.